src/Compilers/CSharp/Test/Emit3/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -9531,4 +9531,83 @@ class C : System.Collections.ICollection
             Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithArguments("_").WithLocation(13, 18)
             );
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70046")]
+    public void PointerType_ListPattern()
+    {
+        var source = """
+            unsafe class C
+            {
+                void M()
+                {
+                    void* v = null;
+                    if (v is [])
+                    {
+                    }
+                }
+            }
+            """;
+        var compilation = CreateCompilation(source, options: TestOptions.UnsafeReleaseDll);
+        compilation.VerifyDiagnostics(
+            // (6,18): error CS8985: List patterns may not be used for a value of type 'void*'. No suitable 'Length' or 'Count' property was found.
+            //         if (v is [])
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[]").WithArguments("void*").WithLocation(6, 18),
+            // (6,18): error CS0518: Predefined type 'System.Index' is not defined or imported
```

Comment:
>// (6,18): error CS0518: Predefined type 'System.Index' is not defined or imported [](http://example.com/codeflow?start=12&length=82)

Let's also test scenario with a target framework that defines the API.

Reply:
added.